lib/shipping_agent/github/notification.rb

```diff
@@ -20,6 +21,34 @@ def update(status, description, deploy)
 
       private
 
+      # Github has an option to do this automaticly, but it only works
+      # for non-production environments.
+      def mark_old_deployments_inactive(deploy)
+        active_old_deployments(deploy).each { |deployment| mark_inactive(deployment) }
+      end
+
+      def active_old_deployments(current)
+        old_deployments(current).reject do |deployment|
+          current.url == deployment.url ||
+            deployment.rels[:statuses].get.data.any? { |s| s.state == "inactive" }
```

Comment:
I'm not sure I understand this bit.... can you elaborate on what's happening here please?

Also, these two methods(`active_old_deployments` and `old_deployments`) don't really seem like `Notification` concerns, should these be on `Deploy`?